src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ReadOnlyTensorSpan.cs

```diff
@@ -395,19 +364,19 @@ internal ReadOnlyTensorSpan(ref T reference, scoped ReadOnlySpan<nint> lengths,
         /// <summary>
         /// The number of items in the span.
         /// </summary>
-        public nint FlattenedLength => _flattenedLength;
+        public nint FlattenedLength => _shape.FlattenedLength;
 
         /// <summary>
         /// Gets a value indicating whether this <see cref="TensorSpan{T}"/> is empty.
         /// </summary>
         /// <value><see langword="true"/> if this span is empty; otherwise, <see langword="false"/>.</value>
-        public bool IsEmpty => _flattenedLength == 0;
+        public bool IsEmpty => _shape.FlattenedLength == 0;
```

Comment:
nit: This could be `=> _shape.IsEmpty`, which ensures all `IsEmpty` checks can use the same logic (which makes improving it in the future, if desired, easier